code/modules/reagents/chemistry/holder.dm

```diff
@@ -685,138 +731,376 @@
+	if(QDELETED(target))
+		CRASH("transfer_reactions() had a [target] ([target.type]) passed to it when it was set to qdel, or it isn't a reagents datum.")
+	if(!reaction_list)
+		return
+	for(var/reaction in reaction_list)
+		var/datum/equilibrium/reaction_source = reaction
+		var/exists = FALSE
+		for(var/reaction2 in target.reaction_list) //Don't add duplicates
+			var/datum/equilibrium/reaction_target = reaction2
+			if(reaction_source.reaction.type == reaction_target.reaction.type) 
+				exists = TRUE
+		if(exists)
+			continue
 
+		var/datum/equilibrium/new_E = new (reaction_source.reaction, target)
+		if(new_E.to_delete)//failed startup checks
+			qdel(new_E)
+		else
+			LAZYADD(target.reaction_list, new_E)
+	target.previous_reagent_list = LAZYLISTDUPLICATE(previous_reagent_list)
+	target.is_reacting = is_reacting
+	
+
+///Checks to see if the reagents has a difference in reagents_list and previous_reagent_list (I.e. if there's a difference between the previous call and the last)
+/datum/reagents/proc/has_changed_state()
+	var/total_matching_reagents = 0
+	for(var/R in previous_reagent_list)
+		if(has_reagent(R))
+			total_matching_reagents++
+	if(total_matching_reagents != reagent_list.len)
+		return TRUE
+	return FALSE
+
+/datum/reagents/proc/update_previous_reagent_list()
+	LAZYNULL(previous_reagent_list)
+	for(var/r in reagent_list)
+		var/datum/reagent/R = r
+		LAZYADD(previous_reagent_list, R.type)
+
+///Old reaction mechanics, edited to work on one only
+///This is changed from the old - purity of the reagents will affect yield
+/datum/reagents/proc/instant_react(datum/chemical_reaction/selected_reaction)
+	var/list/cached_required_reagents = selected_reaction.required_reagents
+	var/list/cached_results = selected_reaction.results
+	var/datum/cached_my_atom = my_atom
+	var/multiplier = INFINITY
+	for(var/B in cached_required_reagents)
+		multiplier = min(multiplier, round(get_reagent_amount(B) / cached_required_reagents[B]))
+
+	if(multiplier == 0)
+		return FALSE
+	var/sum_purity = 0
+	for(var/B in cached_required_reagents)
+		var/datum/reagent/R = has_reagent(B)
+		sum_purity += R.purity
+		remove_reagent(B, (multiplier * cached_required_reagents[B]), safety = 1)
+	sum_purity /= cached_required_reagents.len
+
+	for(var/P in selected_reaction.results)
+		multiplier = max(multiplier, 1) //this shouldn't happen ...
+		var/yield = (cached_results[P]*multiplier)*sum_purity
+		SSblackbox.record_feedback("tally", "chemical_reaction", yield, P)
+		add_reagent(P, yield, null, chem_temp, sum_purity)
+
+	var/list/seen = viewers(4, get_turf(my_atom))
+	var/iconhtml = icon2html(cached_my_atom, seen)
+	if(cached_my_atom)
+		if(!ismob(cached_my_atom)) // No bubbling mobs
+			if(selected_reaction.mix_sound)
+				playsound(get_turf(cached_my_atom), selected_reaction.mix_sound, 80, TRUE)
+
+			for(var/mob/M in seen)
+				my_atom.audible_message(M, "<span class='notice'>[iconhtml] [selected_reaction.mix_message]</span>")
```

Comment:
```suggestion
			for(var/M in seen)
				my_atom.audible_message(M, "<span class='notice'>[iconhtml] [selected_reaction.mix_message]</span>")
```
No need to type check. Only mobs can see.

Reply:
Done for both.